tests/ssg_test_suite/rule.py

```diff
@@ -493,7 +493,8 @@ def copy_of_datastream(self, new_filename=None):
         os.unlink(new_filename)
 
     def _change_variable_value(self, varname, value):
-        _, xslt_filename = tempfile.mkstemp(prefix="xslt-change-value", dir="/tmp")
+        descriptor, xslt_filename = tempfile.mkstemp(prefix="xslt-change-value", dir="/tmp")
+        os.close(descriptor)
```

Comment:
How about using the already open descriptor to write the `template` into it, instead of opening the file again in lines 499~500.

Reply:
We can, but `mkstemp` returns a low-level file descriptor (not file object which is expected by `write()` function), so we will need to use `os.write` and convert the string which we will be writing into the file to bytes.

Reply:
Ahh, okay, overlooked that detail.

Reply:
@matusmarhefka So, the file is opened with `open()` and in line 500 we do `fp.write()`. 
And the docs say that the descriptor should be the same as if an `open()` was done:
https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp

Reply:
```
mkstemp() returns a tuple containing an OS-level handle to an open file (as would be returned by os.open()) and the absolute pathname of that file, in that order.
```
So not `open`, but `os.open`, which means only write from the os module will work (the last in example), but we would need to convert string to bytes:
```
>>> import os
>>> from tempfile import mkstemp
>>> fd, name = mkstemp()
>>> str = """test"""
>>> fd.write(str)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'int' object has no attribute 'write'
>>> os.write(fd, str)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: a bytes-like object is required, not 'str'
>>> os.write(fd, str.encode())
4
```
I don't have strong opinion, but I would rather stick to the original code as it doesn't require conversion to bytes.

Reply:
No problem, thanks for the review.